data-processing/entities/definitions/upload.py

```diff
@@ -1,86 +1,265 @@
-from typing import Optional, cast
+import logging
+from collections import defaultdict
+from typing import Dict, List, Optional, Tuple, cast
 
-from common.types import (
-    BoundingBox,
-    EntityInformation,
-    EntityReference,
-    PaperProcessingResult,
-)
-from common.upload_entities import upload_entities
+from common import file_utils
+from common.colorize_tex import EntityId
+from common.models import EntityData as EntityDataModel
+from common.types import (BoundingBox, Context, EntityData, EntityReference,
+                          EntityRelationships, EntityUploadInfo, MathML,
+                          PaperProcessingResult, SerializableEntity, Symbol)
+from common.upload_entities import (fetch_existing_entities, make_data_models,
+                                    make_relationship_models, upload_entities)
+from entities.symbols.upload import sid
 
-from .types import Definition, TermReference
+from .types import Definiendum, TermReference
 
 
 def upload_definitions(
     processing_summary: PaperProcessingResult, data_version: Optional[int]
 ) -> None:
+    upload_term_definitions(processing_summary, data_version)
+    upload_symbol_definitions(processing_summary, data_version)
 
+
+TermName = str
+
+
+def is_textual_term(entity: SerializableEntity) -> bool:
```

Comment:
Same question with the previous one: Do we treat terms and definiendums different from symbols?

Reply:
yup, we sure do! As mentioned above, we don't have to detect appearnaces of symbols in this step of the pipeline, as they're already detected in an earlier stage of the pipeline. So this function here is used to filter out symbols, so that only appearances of terms and abbreviations are located.

Would it be helpful if I added a comment about this to the code?

Reply:
Yeah, it would be helpful to make a short note about what you described here!

Reply:
Done! This clarification will be added as a comment into the body of the `upload_definitions` function right before the calls to both `upload_term_definitions` and `upload_symbol_definitions`